### DISCUSSION TITLE
File formats to cache raster data?

### DISCUSSION THREAD

**Discussion by advythr** [earthlab-education/Earth-Analytics-2023-01-Intro#676] [! GitHub help]
Hi all,

For my final project, I am working with a large raster dataset which is currently merged in an a xarray Data Array object. I want to cache this object so it does not need to be regenerated every time I close and run my script.

Is there an optimal file format to write this data array to, in order to cache it? Interested to know everyone's thoughts!

**Comment by NobodyLivesHere** [comment]
Would probably want to save it as a .tif with a name indicating it has been merged like
```
local_path = os.path.join('folder here', 'raster-merged'.tif)
with open(local_path, 'wb') as file:
         file.write('ARRAY_HERE')

```

and then add some logic sot that it only executes if there isn't a cached file already present

**Comment by misterskye** [comment]
Ok--very basic, but wow am I having issues getting the cached files to my computer directories.

  I converted and saved the merged Soil ph arrary using:  
                                                                      **ph_file_path = os.path.join(ph_dir, 'lmng_soil_ph')
                                                                      ph_da.rio.to_raster(ph_file_path, driver='GTiff')**              

Then I used this code to open the file:  _rxr.open_rasterio(ph_file_path)_   and it appears the array is different and if I try to create a plot I get an error.

_DataError: xarray DataArray does not define a name and Dataset does not define a default value dimension. Give the DataArray a name or supply an explicit vdim.

XArrayInterface expects gridded data, for more information on supported datatypes see http://holoviews.org/user_guide/Gridded_Datasets.html_

I'm guessing it has to do with the "driver='GTiff'" , but when I didn't apply that code I would get an error.  

Just to note that I tried this method with the SRTM data as well...It saved, but now unable to use xrspatial.

Thanks!!
